From 37b6588e66574a38ec49c8cd95c416ae7a40a5f5 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Sat, 20 Jul 2024 13:42:05 +0200
Subject: [PATCH] CMakeLists.txt: fix buildpaths issue

* use -L to avoid full paths in line control directives:
  libpbnjson-native/2.15.0-16/git $ grep tmp-glibc ../build/src/pbnjson_c/selectors/sele                ectors_grammar_l.c

  #line 1 "/OE/build/luneos-styhead/tmp-glibc/work/x86_64-linux/libpbnjson-native/2.15.0-16/build/src/pbnjson_c/selectors/seletors_grammar_l.c"c
  #line 3 "/OE/build/luneos-styhead/tmp-glibc/work/x86_64-linux/libpbnjson-native/2.15.0-16/build/src/pbnjson_c/selectors/selectors_grammar_l.c"

* as detected by buildpaths QA check:

ERROR: libpbnjson-clang-2.15.0-16-r5 do_package_qa: QA Issue: File /usr/src/debug/libpbnjson-clang/2.15.0-16/src/pbnjson_c/selectors/selectors_grammar_l.c in package libpbnjson-clang-src contains reference to TMPDIR

ERROR: libpbnjson-2.15.0-16-r17 do_package_qa: QA Issue: File /usr/src/debug/libpbnjson/2.15.0-16/src/pbnjson_c/selectors/selectors_grammar_l.c in package libpbnjson-src contains reference to TMPDIR

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Pending [reported in WRP-10884 still waiting for component owner to work]

 src/pbnjson_c/selectors/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pbnjson_c/selectors/CMakeLists.txt b/src/pbnjson_c/selectors/CMakeLists.txt
index e5e3e20..a657c8e 100644
--- a/src/pbnjson_c/selectors/CMakeLists.txt
+++ b/src/pbnjson_c/selectors/CMakeLists.txt
@@ -24,7 +24,7 @@ set(LEMON_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/selectors_grammar_y.c)
 # ---------- FLEX LEXER GENERATOR ------------
 add_custom_command(
 	OUTPUT ${FLEX_OUTPUT}
-	COMMAND ${FLEX_EXECUTABLE} -o ${FLEX_OUTPUT}
+	COMMAND ${FLEX_EXECUTABLE} -L -o ${FLEX_OUTPUT}
 	        ${CMAKE_CURRENT_SOURCE_DIR}/selectors_grammar_l.l
 	DEPENDS ${LEMON_OUTPUT}
 	        ${CMAKE_CURRENT_SOURCE_DIR}/selectors_grammar_l.l
